### DISCUSSION TITLE
Domain Layer vs Application Layer

### DISCUSSION THREAD

**Discussion by vmiokovic** [jasontaylordev/CleanArchitecture#397] [Ideas]
Hi,

I have a quick question thats been bothering me for a while ;)

So I think I know what every layer is for and what goes into each but if we wont have more than one application layer (more than one system) is it ok to make domain and application layers one and the same and just call it core? Or do we always want to have those two separated?

Thank you.

R,
Vid

**Comment by CraigComeOnThisNameCantBeTaken** [comment]
If you have no application layer you will be using infrastructure interfaces in your domain. Now your domain is being watered down by concepts which are not core to the reason your app exists.

But at the end of the day its all about you and what you need.

**Comment by hamidrezahanifidev** [comment]
So the idea behind dividing the core into application and domain layer is to separate the enterprise logic and types (domain layer) from business application and types (application layer). The enterprise logic could be shared across many systems but the business logic is specific to your system only. There is no must in doing these if you don't need. you can actually find the project with only one core layer by checking ardalis/CleanArchitecture

**Comment by jasontaylordev** [comment]
It's a valid approach, check out - https://github.com/dotnet-architecture/eShopOnWeb and see the ApplicationCore project.
